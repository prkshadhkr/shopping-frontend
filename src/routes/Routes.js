import React from 'react';
import { Route, Switch } from 'react-router-dom';
import AuthRoute from './AuthRoute';
import Login from '../user/Login';
import Profile from '../user/Profile';
import Home from '../screens/Home';
import Products from '../screens/Products';
import NewProduct from '../screens/NewProduct';
import ProductDetails from '../screens/ProductDetails';
import Cart from '../screens/Cart';
import Review from '../screens/Review';


const Routes = ({ setToken }) => {
  return (
    <div>
      <Switch>

        <Route exact path="/">
          <Home />
        </Route>

        <Route exact path="/login">
          <Login setToken={setToken} />
        </Route>

        <AuthRoute exact path="/profile">
          <Profile />
        </AuthRoute>

        <AuthRoute exact path="/products">
          <Products />
        </AuthRoute>
     
        <AuthRoute exact path="/products/new">
          <NewProduct />
        </AuthRoute>

        <AuthRoute exact path="/products/:id">
          <ProductDetails />
        </AuthRoute>

       <AuthRoute exact path="/products/:id/reviews">
          <Review />
       </AuthRoute>
       
       <AuthRoute exact path="/cart">
         <Cart />
       </AuthRoute>

     </Switch>
    </div>
  )
}

export default Routes;