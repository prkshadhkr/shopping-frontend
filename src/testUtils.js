import React from  'react';
import UserContext from './user/UserContext';

const demoUser = {
  username: "testing",
  first_name: "testing",
  last_name: "testing",
  email: "testing@gmail.com",
  password: "testing",
  is_admin: false
};

/**
 * user is destructured from userContext in Profilejs as
 * const { user, setUser } = useContext(UserContext);
 */
const UserProvider = ({ children, user = demoUser }) => (
  <UserContext.Provider value={{user}}>
    {children}
  </UserContext.Provider>
)

export { UserProvider };