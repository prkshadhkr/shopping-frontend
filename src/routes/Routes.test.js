import React from 'react';
import { render } from '@testing-library/react';
import Routes from './Routes';
import { MemoryRouter } from 'react-router';
import { UserProvider } from '../testUtils';

it ('renders without crashing', function() {
  render (
    <MemoryRouter>
      <UserProvider>
        <Routes />
      </UserProvider>
    </MemoryRouter>
  )
});

it ('should match the snapshot', function() {
  const { asFragment } = render (
    <MemoryRouter>
      <UserProvider>
        <Routes />
      </UserProvider>
    </MemoryRouter>
  )
  expect(asFragment()).toMatchSnapshot();
});

it ('should match the snapshot if user logged out', function() {
  const { asFragment } = render (
    <MemoryRouter>
      <UserProvider user={null}>
        <Routes />
      </UserProvider>
    </MemoryRouter>
  )
  expect(asFragment()).toMatchSnapshot();
});