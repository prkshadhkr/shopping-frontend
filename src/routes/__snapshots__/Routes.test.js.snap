// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Home d-flex justify-content-center"
    >
      <div>
        <h1
          class="d-flex justify-content-center"
        >
          Shopping
        </h1>
        <p
          class="d-flex justify-content-center font-weight-light"
        >
          Onestop shop: Checkout our great products!
        </p>
        <h5
          class="d-flex justify-content-center"
        >
          Welcome back testing!
        </h5>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should match the snapshot if user logged out 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Home d-flex justify-content-center"
    >
      <div>
        <h1
          class="d-flex justify-content-center"
        >
          Shopping
        </h1>
        <p
          class="d-flex justify-content-center font-weight-light"
        >
          Onestop shop: Checkout our great products!
        </p>
        <div
          class="Login d-flex justify-content-center"
        >
          <a
            class="btn btn-outline-primary"
            href="/login"
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
